### PR TITLE
[aomp-amdgpu] Standalone build

### LINKED ISSUE #280: [aomp-amdgpu] Functional clang for OpenMP offloading on Radeon GPUs
This issue accompanies a series of pull requests that bring `aomp-amdgpu` on Arch Linux:

## General remarks

The Arch Linux packages use the same cmake flags as the upstream build scripts. This means:

* `CMAKE_BUILD_TYPE=Release` for all packages,
* Configure compiler and libraries with `RPATH` pointing to `/opt/rocm/aomp/lib`. This is needed to pick up the right LLVM and OpenMP libraries.
* Currently, only the `C/C++` backend works. `flang` does not, see below.
* The examples (for OpenMP and HIP) from the `aomp` repo run successfully.
* The `aomp` `clang` can also compile HIP code.
* `clang` from `aomp-llvm` can be used as a backend for `hipcc` by setting the environment variable `HIP_CLANG_PATH`:
```
HIP_CLANG_PATH=/opt/rocm/aomp/bin hipcc ...
```

## A simple example
```C
#include <stdio.h>
#include <omp.h>

int main(int argc, char *argv[])
{
    #pragma omp target
    printf("I'm running on the %s.\n", omp_is_initial_device() ? "host" : "accelerator");

    return 0;
}
```
Compile and run:
```bash
/opt/rocm/aomp/bin/clang -fopenmp -fopenmp-targets=amdgcn-amd-amdhsa -Xopenmp-target=amdgcn-amd-amdhsa -march=$(/opt/rocm/aomp/bin/mygpu) simple.c -o simple
./simple
```
Expected output:
```
I'm running on the accelerator.
```

## The metapackage: `aomp-amdgpu` #273 
The current package is split into several single packages; `aomp-amdpu` is now a metapackage similiar to `rocm-dev` and `rocm-libs`

## The backend: `aomp-llvm` #274 
Yet another `LLVM/clang`. This package is not identical with `llvm-amdgpu`.

## The runtime: `aomp-openmp` #276 
Runtime libraries for OpenMP both for host and device calls. Is built twice, once in release, then in debug mode.

## Fortran support: `aomp-flang` `aomp-pgmath` `aomp-flang-runtime` #277 #278 #279 
These packages compile fine but I get a linker error whenever I try to build a fortran program:
```bash
/usr/bin/ld: /opt/rocm/aomp/bin/../lib/libflang.so: undefined reference to `norm2_real8_'
/usr/bin/ld: /opt/rocm/aomp/bin/../lib/libflang.so: undefined reference to `stime'
/usr/bin/ld: /opt/rocm/aomp/bin/../lib/libpgmath.so: undefined reference to `__mth_i_dceil'
/usr/bin/ld: /opt/rocm/aomp/bin/../lib/libpgmath.so: undefined reference to `__mth_i_floor'
/usr/bin/ld: /opt/rocm/aomp/bin/../lib/libflang.so: undefined reference to `norm2_real4_i8_'
/usr/bin/ld: /opt/rocm/aomp/bin/../lib/libpgmath.so: undefined reference to `__mth_i_dfloor'
/usr/bin/ld: /opt/rocm/aomp/bin/../lib/libflang.so: undefined reference to `norm2_real4_'
/usr/bin/ld: /opt/rocm/aomp/bin/../lib/libflang.so: undefined reference to `norm2_real8_i8_'
```

### PR DESCRIPTION
**Pull request by tpkessler**
Resolves #280. Following the suggestions upstream, `aomp-amdgpu` is now built in standalone mode. This means that most of the `rocm-dev` packages are included in `aomp-amdgpu`.

### CODE REVIEW

File: aomp-amdgpu/PKGBUILD [hunk 1]
```diff
@@ -1,23 +1,53 @@
...
-build() {
-  cd ${srcdir}/aomp-rocm-$pkgver
-  nohup make
+url='https://github.com/ROCm-Developer-Tools/aomp'
+license=('Apache')
+depends=(z3 numactl pciutils libelf libffi)
+makedepends=(git cmake python)
+source=("${pkgname}-${pkgver}.tar.gz::$url/archive/rel_$_pkgver.tar.gz"
+        'disable_ocl_tests.patch'
+        'adjust_rpath.patch')
```
Comment (acxz) [inline aomp-amdgpu/PKGBUILD:16]: I believe these patches are not included in the commit. Also if they can be contributed upstream that would be ideal.

Comment (tpkessler) [reply aomp-amdgpu/PKGBUILD:16]: I've added the missing patch files. The first patch removes unnecessary tests, the second changes RPATH from `installdir`, which is `$pkdir/opt/rocm/aomp`, to `/opt/rocm/aomp`. This is only needed for packaging.

Comment (acxz) [reply aomp-amdgpu/PKGBUILD:16]: I think the patches were added in the wrong directory.

I see kk

Comment (tpkessler) [reply aomp-amdgpu/PKGBUILD:16]: Thanks. I've fixed it.

**Pull request merged**